samples/bookinfo/kube/route-rule-all-v1.yaml

```diff
@@ -1,6 +1,105 @@
 apiVersion: config.istio.io/v1alpha2
 kind: RouteRule
 metadata:
+  name: ingress-deny-all
+spec:
+  destination:
+    name: productpage
+  match:
+    source:
+      labels:
+        istio: ingress
+  precedence: 1
+  route:
+  - weight: 100
+  httpFault:
+    abort:
+      percent: 100
+      httpStatus: 403
+---
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: ingress-products-api
+spec:
+  destination:
+    name: productpage
+  match:
+    source:
+      labels:
+        istio: ingress
+    request:
+      headers:
+        uri:
+          prefix: /api/v1/products
+  precedence: 2
+  route:
+  - labels:
+      version: v1
+---
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: ingress-productpage-logout
+spec:
+  destination:
+    name: productpage
+  match:
+    source:
+      labels:
+        istio: ingress
+    request:
+      headers:
+        uri:
+          exact: /logout
+  precedence: 3
+  route:
+  - labels:
+      version: v1
+---
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: ingress-productpage-login
+spec:
+  destination:
+    name: productpage
+  match:
+    source:
+      labels:
+        istio: ingress
+    request:
+      headers:
+        uri:
+          exact: /login
+  precedence: 4
+  route:
+  - labels:
+      version: v1
+---
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: ingress-productpage-landing
+spec:
+  destination:
+    name: productpage
+  match:
+    source:
+      labels:
+        istio: ingress
+    request:
+      headers:
+        uri:
+          exact: /productpage
+  precedence: 5
```

Comment:
I think the rules like these that have non-overlapping match clauses could all be `precedence: 2`, unless for performance reasons we want to evaluate the rule in a specific order, based on how often it's called. Is that what you are doing here? If so, even in that case I think that login and logout could both be the same precedence value, and /api/v1/products should be higher precedence.

Reply:
same precedence sounds fine. I was trying to avoid non determinism in pilot, but I think we would be handling it anyway, for things with same precedence. will change it once the tests pass.